src/plugins/data/public/ui/query_editor/language_selector.tsx

```diff
@@ -64,25 +84,56 @@ export const QueryLanguageSelector = (props: Props) => {
       )?.label as string) ?? languageOptions[0].label,
   };
 
-  const handleLanguageChange = (newLanguage: EuiComboBoxOptionOption[]) => {
-    const queryLanguage = newLanguage[0].value as string;
-    props.onSelectLanguage(queryLanguage);
-    uiService.Settings.setUserQueryLanguage(queryLanguage);
+  const handleLanguageChange = (newLanguage: any) => {
+    props.onSelectLanguage(newLanguage);
+    uiService.Settings.setUserQueryLanguage(newLanguage);
   };
 
-  uiService.Settings.setUserQueryLanguage(props.language);
+  // language selector at the footer of the query bar
+  const button = (
+    <EuiButtonEmpty
+      iconSide="right"
+      onClick={onButtonClick}
+      style={{
+        fontSize: 'small',
+        height: '20px',
+      }}
+    >
+      {selectedLanguage.label}
+    </EuiButtonEmpty>
+  );
+  const getIconType = (language: string) => {
+    return language === selectedLanguage.label ? 'check' : 'empty';
```

Comment:
how come? because it's buggy? or we want it to have the check?

Reply:
when user clicks the language selector button, this is rendering the check icon next to whatever language being selected. 
![Screenshot 2024-07-19 at 1 58 43 AM](https://github.com/user-attachments/assets/3a114d6c-6c47-4125-8f2e-e8692a58a20e)

Reply:
also made it a one-liner instead of a new function